app/src/pages/inside/settingsPage/settingsTabs/defectTypesTab/defectTypesGroup/defectTypesGroup.jsx

```diff
@@ -0,0 +1,345 @@
+    projectId: activeProjectSelector(state),
+    subTypes: defectTypesSelector(state),
+    config: projectConfigSelector(state),
+  }),
+  {
+    showNotification,
+    showModalAction,
+    hideModalAction,
+    deleteDefectTypeAction,
+  },
+)
+@injectIntl
+export class DefectTypesGroup extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    showNotification: PropTypes.func,
+    showModalAction: PropTypes.func,
+    hideModalAction: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    projectId: PropTypes.string.isRequired,
+    initialize: PropTypes.func,
+    title: PropTypes.string.isRequired,
+    group: PropTypes.string.isRequired,
+    subType: PropTypes.array.isRequired,
+    cx: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    intl: {},
+    showNotification: () => {},
+    showModalAction: () => {},
+    hideModalAction: () => {},
+    projectId: '',
+    fetchProjectAction: () => {},
+    initialize: () => {},
+    handleSubmit: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      defectTypes: this.props.subType,
+      isInEdit: [],
+      isNewInEdit: false,
+      newColor: this.props.subType[0].color,
+    };
+  }
+
+  onDeleteHandler = (defectType) => (data) => {
+    const { formatMessage } = this.props.intl;
+    this.props.showModalAction({
+      id: 'defectTypeDeleteModal',
+      data: {
+        title: formatMessage(Messages.deleteModalTitle),
+        message: formatMessage(Messages.deleteModalConfirmationText, {
+          defectType: defectType.longName,
+          defaultDefectGroup: formatMessage(Messages[defectType.typeRef.toLowerCase()]),
+        }),
+        submitText: formatMessage(COMMON_LOCALE_KEYS.DELETE),
+        cancelText: formatMessage(COMMON_LOCALE_KEYS.CANCEL),
+        onSubmit: this.deleteDefectType(defectType),
+        hideModalAction: this.props.hideModalAction,
+        data,
+      },
+    });
+  };
+
+  setNewColor = (color) => {
+    if (color !== this.state.newColor) {
+      this.setState({ newColor: color });
+    }
+  };
+
+  setNewEditMode = (editMode) => this.setState({ isNewInEdit: editMode });
+
+  setDefectTypeEditMode = (defectType) => (editMode) =>
+    this.setState({
+      newColor: defectType.color,
+      isInEdit: editMode
+        ? this.state.isInEdit.concat(defectType.locator)
+        : this.state.isInEdit.filter((locator) => locator !== defectType.locator),
+    });
+
+  MAX_DEFECT_SUBTYPES_SIZE = 15;
+
+  showSuccessNotification = (message) =>
+    this.props.showNotification({
+      message: this.props.intl.formatMessage(message),
+      type: NOTIFICATION_TYPES.SUCCESS,
+    });
+
+  showErrorNotification = (message) =>
+    this.props.showNotification({
+      message: this.props.intl.formatMessage(message),
+      type: NOTIFICATION_TYPES.ERROR,
+    });
+
+  deleteDefectType = (deletedDefectType) => () =>
+    this.deleteDefectTypeRequest(deletedDefectType, (deletedLocator) => {
```

Comment:
deleteDefectTypeRequest
deleteDefectType

Can it be a single function? Or maybe these two can have more meaningful names?